### PR TITLE
Support reading Array(Record) into flatten nested table in Avro

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support reading Array(Record) into flatten nested table in Avro

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by Avogar** [comment]
See comment https://github.com/ClickHouse/ClickHouse/pull/37525#issuecomment-1208015030

File: src/Processors/Formats/Impl/AvroRowInputFormat.cpp [hunk 1]
```diff
@@ -631,6 +665,42 @@ AvroDeserializer::Action AvroDeserializer::createAction(const Block & header, co
         }
         return AvroDeserializer::Action::unionAction(branch_actions);
     }
+    else if (node->type() == avro::AVRO_ARRAY)
+    {
+        /// If header doesn't have column with current_path name and node is Array(Record),
+        /// check if we have a flattened Nested table with such name.
+        Names nested_names = Nested::getAllNestedColumnsForTable(header, current_path);
+        auto nested_avro_node = node->leafAt(0);
```
Comment (vdimir) [inline src/Processors/Formats/Impl/AvroRowInputFormat.cpp:673]: Can't we have empty node here, so is it safe to call `leafAt(0)` without checking `node->leaves()`?

**Review by vdimir — APPROVED** [review]

File: src/Processors/Formats/Impl/AvroRowInputFormat.cpp (hunk 1 above)
Comment (Avogar) [reply src/Processors/Formats/Impl/AvroRowInputFormat.cpp:673]: We checked that node type is Array, so it should have one leaf

Comment (Avogar) [reply src/Processors/Formats/Impl/AvroRowInputFormat.cpp:673]: Hm, but it's interesting if it's possible to corrupt data in this way

Comment (Avogar) [reply src/Processors/Formats/Impl/AvroRowInputFormat.cpp:673]: Tried to create such corrupted file, but without success, reading such file failes in avro library before this code

**Pull request merged**